pkg/plugin/aggregation/aggregator.go

```diff
@@ -110,47 +132,66 @@ func (a *Aggregator) isResultDuplicate(result *plugin.Result) bool {
 	return ok
 }
 
-// HandleHTTPResult is called every time the HTTP server gets a well-formed
-// request with results. This method is responsible for returning with things
-// like a 409 conflict if a node has checked in twice (or a 403 forbidden if a
-// node isn't expected), as well as actually calling handleResult to write the
-// results to OutputDir.
-func (a *Aggregator) HandleHTTPResult(result *plugin.Result, w http.ResponseWriter) {
+// processResult is the centralized location for result processing. It is thread-safe
+// and checks for whether or not the result should be excluded due to be either
+// unexpected or a duplicate. Errors returned via this method will be of the type
+// *httpError so that HTTP servers can respond appropriately to clients.
+func (a *Aggregator) processResult(result *plugin.Result) error {
 	a.resultsMutex.Lock()
 	defer a.resultsMutex.Unlock()
 
 	resultID := result.ExpectedResultID()
 
 	// Make sure we were expecting this result
 	if !a.isResultExpected(result) {
-		http.Error(
-			w,
-			fmt.Sprintf("Result %v unexpected", resultID),
-			http.StatusForbidden,
-		)
-		return
+		return &httpError{
+			err:  fmt.Errorf("result %v unexpected", resultID),
+			code: http.StatusForbidden,
```

Comment:
A `403` seems out of place here for this type of error.

Reply:
Agree, looking at docs that seems like it is supposed to be auth related, not content like this. Not 100% sure of the right response though but that is for another PR; this just moved the code and didnt chose it here. Maybe https://httpstatuses.com/422 would be most appropriate.